src/lex.c

```diff
@@ -163,6 +196,12 @@ token_t *readTokensFromFile(FILE *file){
 		type_t type = getTokenType((char *) buf);
 		addToken(buf, type);
 	}
+
+	if (c == EOF){
+		buf[0] = c;
+		addToken(buf, tok_bofeof);
+	}
```

Comment:
Wenn man aus der vorherigen Schleife ausgebrochen ist, kann man sich sicher sein, dass ``c == EOF`` ist. Ich würde den Token bedingungslos anhängen.

```suggestion
	addToken("\n", tok_bofeof);
```